### DISCUSSION TITLE
Proposal: .= operator

### DISCUSSION THREAD

**Discussion by monke8555** [dotnet/csharplang#4763] [Language Ideas] [answered]
I once had a problem doing this for a long variable:
```
string SomeLongVarName = Value;
SomeLongVarName = SomeLongVarName.Replace("hello", "Hello,");
```
So I had an idea, that this could be better.  
So my proposal is a .= operator.  
With that, the above could be done like this.  
```
SomeLongVarName .= Replace("hello", "Hello,");
```
This could make for smaller code files in bigger projects that need much string manipulation.  

This is probably not worth the effort, but I had the idea and thought I'd share it here.

**Comment by En3Tho** [comment]
This syntax has little value as it just saves a few characters vs saves from a mistake and instead it potentially can cause errors as the only thing that distinguishes static call vs instance call is that small dot before the equals sign. Which is contra intuitive. I don't believe pressing ctrl-c ctrl-v requires any kind of special syntax.

**Reply by monke8555** [reply]
> This is probably not worth the effort, but I had the idea and thought I'd share it here.

Ya I know

**Comment by Joe4evr** [comment, marked as answer]
Duplicate of #4245

**Reply by Unknown6656** [reply]
and a duplicate of https://github.com/dotnet/roslyn/issues/9134
